src/distribution/distribution.jl

```diff
@@ -3,21 +3,45 @@ using StaticArrays
 using LinearAlgebra
 import ..TINY
 
-abstract type AdaptiveMap end
+# abstract type AdaptiveMap end
 
-"""
-    abstract type Variable end
+# """
+#     abstract type Variable end
 
-Abstract Type of all variable pools. 
-"""
-abstract type Variable end
+# Abstract Type of all variable pools. 
+# """
+# abstract type Variable end
+# abstract type ScalarVariable{T} end
+abstract type AbstractVectorVariable{T} <: AbstractVector{T} end
+# struct MatrixVariable <: Variable end
+# struct NonVariable <: Variable end
+
+is_variable(::Type) = false
+is_variable(::Type{<:AbstractVector{T}}) where {T} = true
```

Comment:
Shouldn't the trait `is_variable` be true only for `AbstractVectorVariable` instances, rather than all `AbstractVector{T}`? Would `is_variable(::Type{<:AbstractVectorVariable{T}}) where {T} = true` make more sense?

Reply:
right, that is a bug

Reply:
good catch